src/com/xmartlabs/slackbot/manager/TogglReportManager.kt

```diff
@@ -0,0 +1,100 @@
+package com.xmartlabs.slackbot.manager
+
+import com.slack.api.model.User
+import com.xmartlabs.slackbot.Config
+import com.xmartlabs.slackbot.extensions.toPrettyString
+import com.xmartlabs.slackbot.logger
+import com.xmartlabs.slackbot.model.TogglUserEntryReport
+import com.xmartlabs.slackbot.repositories.ConversationSlackRepository
+import com.xmartlabs.slackbot.repositories.TogglReportRepository
+import com.xmartlabs.slackbot.repositories.UserSlackRepository
+import java.time.Duration
+import java.time.LocalDateTime
+import kotlin.time.ExperimentalTime
+import kotlin.time.toKotlinDuration
+
+@OptIn(ExperimentalTime::class)
+object TogglReportManager {
+    suspend fun sendReport(
+        reportType: ReportType,
+        from: LocalDateTime,
+        to: LocalDateTime,
+    ) {
+        val now = LocalDateTime.now()
+        val users = UserSlackRepository.getUsers()
+            .associateBy { it.profile.email }
+        logger.info("Users: ${users.keys}")
+
+        logger.info("Fetch toggl entries from: $from to $to")
+        TogglReportRepository.getEntriesInWrongFormat(from, to)
+            .filter { (_, untrackedTime) ->
+                reportType == ReportType.MONTHLY ||
+                        untrackedTime >= Config.TOGGL_WEEKLY_REPORT_MIN_UNTRACKED_DURATION_TO_NOTIFY
+            }
+            .mapNotNull { report ->
+                val slackUser = users[report.togglUser.email]
+                if (slackUser == null) {
+                    logger.warn("Slack user not found. Toggl user: ${report.togglUser}")
+                    null
+                } else {
+                    slackUser to report
+                }
+            }
+            .also {
+                val calculationTime = Duration.between(now, LocalDateTime.now())
+                    .toKotlinDuration()
+                logger.info("Users to notify (calculated in $calculationTime)")
```

Comment:
i don't understand this

Reply:
The idea was to track the time that took the report calculation 
```suggestion
                val calculationTime = Duration.between(now, LocalDateTime.now())
                    .toKotlinDuration()
                logger.info("Toogle report generated (generation time $calculationTime)")
```